dubbo-plugin/dubbo-qos/src/main/java/org/apache/dubbo/qos/protocol/QosProtocolWrapper.java

```diff
@@ -102,6 +104,14 @@ private void startQosServer(URL url) {
                     "dubbo.properties or XML/spring-boot configuration.");
                 return;
             }
+            // frameModel of mock url is null
+            FrameworkModel frameworkModel1= url.getOrDefaultFrameworkModel();
+            if (frameworkModel1 != null) {
+                WireProtocol protocol1 = frameworkModel1.getExtensionLoader(WireProtocol.class).getExtension("qos");
+                if(protocol1 != null) {
+                    ((QosWireProtocol) protocol1).SetQosEnable(true);
```

Comment:
This will cause it to be turned on forever.

Reply:
if qosEnable is not true, this logic won't work. and default flag for qoswireProtocol to enable qos detector in pu server is false.